src/files.ts

```diff
@@ -112,16 +113,17 @@ export async function getProjectFiles(rootDir: string = path.join('.', '/'), cal
 
         // check ignore files
         const ignoreMatches = multimatch(filePaths, ignorePatterns, { dot: true });
+        const intersection: string[] = filePaths.filter(file => !ignoreMatches.includes(file));
 
-        // Loop through every file.
-        const files = filePaths
+        // Loop through files that are not ignored
```

Comment:
Looping through every file (and doing all this logic to each file, takes forever). Just loop through non-ignored files to make sure they are valid.